### DISCUSSION TITLE
Poetry run seems to be executing in global environment instead of virtual one

### DISCUSSION THREAD

**Discussion by ifateri** [python-poetry/poetry#6757] [Q&A]
Hello,

I'm using Python for data analysis and have recently switched to Poetry. 

Following the documentation I have initialised a pre-existing directory to set up with poetry. After activating my virtual environment with `source <path to env>/Scripts/activate` I then use `poetry run python scr/script.py` but it fails because a specific module can't be imported (psycopg2). Whilst in the virtual environment `poetry show` return that package as installed, whereas `pip list` in the global environment doesn't. 

As soon as I install psycopg2 in the global environment with `pip install` I can then successfully execute the Python script with `poetry run`. 

I'm not sure whether this is a bug or whether I've misunderstood something. Any help appreciated!

Info:
- Windows 10
- Python 3.8.3
- Poetry 1.2.1 (via recommended installer)

**Comment by finswimmer** [comment]
Hey,

`poetry show` shows all locked dependencies. You have to take care of the color. If the name is printed in red, this package is not installed.

You've said, that you already have activated the venv. What's the output of `pip list` in this environment? Have you run `poetry install` at least once?

fin swimmer

**Comment by ifateri** [comment]
Hello,

Thanks!

I didn't know `poetry show` used colours, those don't appear for me in Git Bash in VSCode (but they do in Git bash). In the virtual environment if I do `poetry show` I see every package in what I think is green and when I do `pip list` inside the virtual environment I see the package. I also did `poetry install` and said nothing to install.

I just removed the package from the global environment again and then hit the same "module not found" error when executing `poetry run` so it really looks to me it's using the global environment? Could it be related to this [issue ](https://github.com/python-poetry/poetry/issues/6495)with `poetry shell`? I have to use `source <full path>/Scripts/activate` to get my virtual environment running. Do I need to supply this to `poetry run` too or something?
